docker-compose.yml

```diff
@@ -0,0 +1,31 @@
+version: '3.8'
+services:
+  clickhouse:
+    image: 'clickhouse/clickhouse-server:${CLICKHOUSE_VERSION-24.3-alpine}'
+    container_name: 'clickhouse-connect-clickhouse-server'
+    ports:
+      - '8123:8123'
+      - '9000:9000'
+    ulimits:
+      nofile:
+        soft: 262144
+        hard: 262144
+    volumes:
+      - './.docker/clickhouse/single_node/config.xml:/etc/clickhouse-server/config.xml'
+      - './.docker/clickhouse/users.xml:/etc/clickhouse-server/users.xml'
+
+  clickhouse_tls:
+    build:
+      context: ./
+      dockerfile: .docker/clickhouse/single_node_tls/Dockerfile
+    container_name: 'clickhouse-connect-clickhouse-server-tls'
+    ports:
+      - '10843:8443'
+      - '10840:9440'
+    ulimits:
+      nofile:
+        soft: 262144
+        hard: 262144
+    volumes:
+      - './.docker/clickhouse/single_node_tls/config.xml:/etc/clickhouse-server/config.xml'
+      - './.docker/clickhouse/single_node_tls/users.xml:/etc/clickhouse-server/users.xml'
```

Comment:
The entire setup is mostly taken from CHJS (with the certificates from this project).